src/util.cpp

```diff
@@ -196,8 +196,15 @@ std::string AbsolutePath(const std::string &filepath) {
       char abs_path[MAX_PATH];
       return GetFullPathNameA(filepath.c_str(), MAX_PATH, abs_path, nullptr)
     #else
-      char abs_path[PATH_MAX];
-      return realpath(filepath.c_str(), abs_path)
+	  char* abs_path_temp = realpath(filepath.c_str(), nullptr);
+	  bool success = abs_path_temp != nullptr;
+	  std::string abs_path;
+	  if(success)
+	  {
+		  abs_path = abs_path_temp;
+		  free(abs_path_temp);
```

Comment:
why not use `abs_path` as before? that could avoid the malloc, and simplify this code.

Reply:
The tricky part is the PATH_MAX. If it is not constant it should not be defined (according POSIX). Causes can be dependency on filesystem and thus path for example. On certain OS (such as QNX which I use) that are fully POSIX compliant this means that PATH_MAX is not defined if you adhere to a standard. The alternative is pathconf() (a lot of work) or this. This is fully POSIX 2008 compliant and portable to any POSIX compliant platform.

[realpath](http://man7.org/linux/man-pages/man3/realpath.3.html)
[limits.h](https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html)